src/Container/ServiceExtensions.php

```diff
@@ -147,7 +150,7 @@ protected function resolveByType(
      * @param class-string $type
      * @param object $service
      * @param Container $container
-     * @param array $extenders
+     * @param list<ExtendingService> $extenders
```

Comment:
Not sure if this is relevant, but can we be sure that `$extenders` always is a list (and not some associative array)? Given that we don't even use the keys, maybe we should just use `ExtendingService[]`?

Reply:
I just kept it aligned with the internal docBlock here: https://github.com/inpsyde/modularity/blob/2c50cb62d09c06d125d0a3ea1a4bedddf977b6da/src/Container/ServiceExtensions.php#L19 and `ExtendingServices` are always added in an int-based array: https://github.com/inpsyde/modularity/blob/master/src/Container/ServiceExtensions.php#L37 (1 Service can have multiple ExtendingServices).

The `ExtendingService[]` is PHPDoc syntax is is equivalent to `array<array-key, ValueType>` in psalm. So in theory it's the same, but `list<ExtendingService>` seems to be stricter because it is `array<int, ExtendingService>`.

I guess both are right, while `list<>` is stricter and more accurate in this case? 🤔  @gmazzap any opinion here?

Reply:
As already said by both of you, it is a private method, Psalm can check all usages, so better go stricter.